src/ui/new-factory.tsx

```diff
@@ -0,0 +1,247 @@
+/**
+ * ui/new-factory.ts
+ * React components for creating a new factory
+ * lgfrbcsgo & Nikolaus - October 2020
+ */
+
+import * as React from "react"
+import { Craftable, isCraftable, ITEMS } from "../items"
+import { values } from "ramda"
+import { ItemSelect } from "./item-select"
+import { useMap } from "./app"
+import { Button, Row, Col, InputNumber } from "antd"
+import { buildFactory } from "../factory"
+import { FactoryGraph } from "../graph"
+import { AppState } from "./app"
+import { FactoryInstruction, generateInstructions, FactoryVisualization } from "./render-factory"
+
+export enum FactoryState {
+    SELECT = "select",
+    COUNT = "count",
+    RENDER = "render",
+}
+
+/**
+ * Properties of the NewFactory component
+ */
+interface NewFactoryProps {
+    /**
+     * Set the parent application state
+     * @param state the AppState
+     */
+    setAppState: (state: AppState) => void
+}
+
+/**
+ * New Factory component
+ * @param props {@link NewFactoryProps}
+ */
+export function NewFactory({ setAppState }: NewFactoryProps) {
+    // all possible craftable items
+    const items = React.useMemo(() => values(ITEMS).filter(isCraftable), [ITEMS])
+    // NewFactory state and factory instructions
+    const [factoryState, setFactoryState] = React.useState<FactoryState>(FactoryState.SELECT)
+    const [factoryInstructions, setFactoryInstructions] = React.useState<FactoryInstruction[]>([])
+    // produced items, industry count, and maintain count
+    const [selection, setSelection] = React.useState<Craftable[]>([])
+    const [industryCount, setIndustryCount] = useMap<Craftable, number>()
+    const [maintainValue, setMaintainValue] = useMap<Craftable, number>()
+    // the FactoryGraph
+    const [factory, setFactory] = React.useState<FactoryGraph>()
```

Comment:
We should probably refactor this into a single state object when we add another `useState`. This is fine for now.